guides/source/routing.md

```diff
@@ -18,51 +18,66 @@ After reading this guide, you will know:
 The Purpose of the Rails Router
 -------------------------------
 
-The Rails router recognizes URLs and dispatches them to a controller's action, or to a Rack application. It can also generate paths and URLs, avoiding the need to hardcode strings in your views.
+The Rails router matches incoming HTTP requests to specific controller actions
+in your Rails application based on the URL path. (It can also forward to a
+[Rack](rails_on_rack.html) application.) The router also generates path and URL
+helpers based on the resources configured in the router.
 
-### Connecting URLs to Code
+### Routing Incoming URLs to Code
 
-When your Rails application receives an incoming request for:
+When your Rails application receives an incoming request, it asks the router to match it to a controller action. For example, take the following incoming request:
```

Comment:
We can match a route to [any rack app](https://guides.rubyonrails.org/routing.html#routing-to-rack-applications), which not necessary is a controller action. I'm not sure which wording to use, though.

Reply:
edited. Yes we do mention Rack in the intro, right above.